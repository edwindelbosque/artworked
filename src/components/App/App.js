import React, { Component } from 'react';
import './App.scss';
import SearchForm from '../SearchForm/SearchForm';
import Container from '../Container/Container';
import ArtworkModal from '../ArtworkModal/ArtworkModal';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import Nav from '../Nav/Nav';

export class App extends Component {
	render() {
		const { results, favorites } = this.props;
		return (
			<main className='App'>
				<Nav />
				<Route exact path='(|search/:id)' render={() => <SearchForm />} />
				<Route path='(|search/:id)' render={() => <Container />} />
				<Route
					path='/(search|favorites)/:id'
					render={({ match }) => {
						console.log(match);
						const id = parseInt(match.params.id);
						const data = [...results, ...favorites];
						const result = data.find(result => result.id === id);
						return <ArtworkModal result={result} match={match.params} />;
					}}
				/>
				<Route render={() => <Redirect to={{ pathname: '/' }} />} />
			</main>
		);
	}
}

export const mapStateToProps = ({ results, favorites }) => ({
	results,
	favorites
});

export default connect(mapStateToProps)(App);
