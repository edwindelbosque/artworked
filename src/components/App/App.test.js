import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<App />);
	});

	it('should match App snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
