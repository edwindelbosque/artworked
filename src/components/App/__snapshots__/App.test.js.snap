// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match App snapshot 1`] = `
<main
  className="App"
>
  <Connect(Nav) />
  <Route
    exact={true}
    path="(|search/:id)"
    render={[Function]}
  />
  <Route
    path="(|search/:id)"
    render={[Function]}
  />
  <Route
    path="/(search|favorites)/:id"
    render={[Function]}
  />
  <Route
    render={[Function]}
  />
</main>
`;
